osu.Game/Beatmaps/Beatmap.cs

```diff
@@ -61,4 +62,15 @@ public class Beatmap : Beatmap<HitObject>
     {
         public new Beatmap Clone() => (Beatmap)base.Clone();
     }
+
+    public static class BeatmapExtensions
+    {
+        public static double CalculateLength(this IBeatmap b)
```

Comment:
why is this an extension method?

Reply:
it feels more better, unless you prefer property, i'm fine with it

Reply:
i don't understand why it feels better. please elaborate.

Reply:
it's just my opinion, but i'm fine with any change you suggest

*just realized extensions are not needed if you can add to the class itself*